src/main/kotlin/apply/application/AssignmentService.kt

```diff
@@ -0,0 +1,50 @@
+package apply.application
+
+import apply.domain.assignment.Assignment
+import apply.domain.assignment.AssignmentRepository
+import apply.domain.evaluationtarget.EvaluationStatus
+import apply.domain.evaluationtarget.EvaluationTarget
+import apply.domain.evaluationtarget.EvaluationTargetRepository
+import apply.domain.mission.MissionRepository
+import apply.domain.mission.getById
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+@Service
+class AssignmentService(
+    private val assignmentRepository: AssignmentRepository,
+    private val missionRepository: MissionRepository,
+    private val evaluationTargetRepository: EvaluationTargetRepository
+) {
+    fun create(missionId: Long, userId: Long, request: CreateAssignmentRequest) {
+        check(!assignmentRepository.existsByMissionIdAndUserId(missionId, userId)) {
+            "이미 제출한 과제물이 존재합니다."
+        }
+
+        val mission = missionRepository.getById(missionId)
+        check(mission.isProgressing) {
+            "제출 불가능한 과제입니다."
+        }
```

Comment:
조건문이 아닌 람다의 중괄호는 한 줄로 줄여도 좋을 것 같아요.

```suggestion
        check(mission.isProgressing) { "제출 불가능한 과제입니다." }
```

Reply:
"중복 제출 과제" check도 함께 반영했습니다.
감사합니다.

``` kotlin
check(!assignmentRepository.existsByMissionIdAndUserId(missionId, userId)) { "이미 제출한 과제물이 존재합니다." }
check(mission.isProgressing) { "제출 불가능한 과제입니다." }
```